datafusion/physical-plan/src/async_func.rs

```diff
@@ -100,6 +100,14 @@ impl AsyncFuncExec {
             input.boundedness(),
         ))
     }
+
+    pub fn async_exprs(&self) -> Vec<Arc<AsyncFuncExpr>> {
+        self.async_exprs.clone()
+    }
+
+    pub fn input(&self) -> Arc<dyn ExecutionPlan> {
+        Arc::clone(&self.input)
+    }
```

Comment:
```suggestion
    pub fn async_exprs(&self) -> &[Arc<AsyncFuncExpr>] {
        &self.async_exprs
    }

    pub fn input(&self) -> &Arc<dyn ExecutionPlan> {
        &self.input
    }
```

To let the caller decide if they want to clone